js/components/src/time-sync/useTimeSync.tsx

```diff
@@ -0,0 +1,56 @@
+import { TriangleAlert } from "lucide-react-native";
+import { useEffect, useRef } from "react";
+import { Platform } from "react-native";
+import { StreamplaceAgent } from "streamplace";
+import { useToast } from "../components/ui/toast";
+import { useUrl } from "../streamplace-store/streamplace-store";
+import { checkClockDrift, syncTimeWithServer } from "./time-sync";
+
+export function useTimeSync() {
+  const url = useUrl();
+  const t = useToast();
+  const hasShownWarning = useRef(false);
+
+  useEffect(() => {
+    const checkTime = async () => {
+      if (Platform.OS !== "web") {
+        return;
+      }
+      try {
+        const agent = new StreamplaceAgent(url);
+        const response = await agent.place.stream.server.getServerTime();
+        const serverTime = response.data.serverTime;
+
+        // always sync with server time
+        syncTimeWithServer(serverTime);
```

Comment:
So technically we can be ever-so-slightly more accurate here by assuming that the server generated its time halfway through the round-trip and compensating for that. It matters in the case where you're on a plane or something with working internet but 5000ms latency
```suggestion
        const agent = new StreamplaceAgent(url);
        const start = Date.getTime();
        const response = await agent.place.stream.server.getServerTime();
        const roundTripLatency = Date.getTime() - start;
        const serverTime = response.data.serverTime;

        // always sync with server time
        syncTimeWithServer(serverTime, roundTripLatency / 2);
```